resources/domain/signer/index.ts

```diff
@@ -1,9 +1,52 @@
-import Signer from '../../../main/signers/Signer'
+import type Signer from '../../../main/signers/Signer'
 
-export function getSignerType (signer: Signer) {
+// in order of increasing priority
+export enum Type {
+  Ring = 'ring',
+  Seed = 'seed',
+  Aragon = 'aragon',
+  Trezor = 'trezor',
+  Ledger = 'ledger',
+  Lattice = 'lattice'
+}
+
+export function getSignerType (typeValue: string) {
+  return Object.values(Type).find(type => type === typeValue)
+}
+
+export function getSignerDisplayType (signer: Signer) {
   return ['ring', 'seed'].includes((signer.type || '').toLowerCase()) ? 'hot' : signer.type
 }
 
 export function isHardwareSigner (type = '') {
   return ['ledger', 'trezor', 'lattice'].includes(type.toLowerCase())
 }
+
+export function getAvailableSigner (signer: Signer, lastSignerType: Type, getSigners: () => Signer[], cb: Callback<Signer>) {
```

Comment:
I like centralizing this business logic but I don't think we need all of it here. a couple things:

- this function isn't async so it doesn't need to take a callback. the one in accounts doesn't really either and that's a bigger refactor but for now let's keep that issue isolated to that method
- some of this logic doesn't apply globally. for example the `isSignerReady` check below only applies when we want to sign something and _not_ when clicking the button at the bottom of the account panel. also any code that calls this can check if the signer if falsy before calling it so we don't need to handle that here. I think the only logic that needs to be in this shared space is the lookup of a disconnected hardware signer